### PR TITLE
Add reply feature to subscription docs

### PR DESCRIPTION
**Pull request by abrennan89**
Fixes https://github.com/knative/docs/issues/4937

### CODE REVIEW

File: docs/eventing/channels/subscriptions.md [hunk 1]
```diff
@@ -74,23 +74,28 @@ For more information about Subscription objects, see
...
+              kind: InMemoryChannel
+              name: <service-name>
+              # Configuration settings for the reply event. This is the event Sink that events replied from the subscriber are delivered to.
           subscriber:
             ref:
               apiVersion: serving.knative.dev/v1
               kind: Service
-              name: <service-name> # Configuration settings for the Subscriber. This is the event Sink that events are delivered to from the Channel.
+              name: <service-name>
+              # Configuration settings for the Subscriber. This is the event Sink that events are delivered to from the Channel.
```
Comment (pierDipi) [inline docs/eventing/channels/subscriptions.md:98]: Can we move the comments on top of the field like:
```yaml
          # Configuration settings for the reply event. This is the event Sink that events replied from the subscriber are delivered to.
          reply:
            ref:
              apiVersion: messaging.knative.dev/v1
              kind: InMemoryChannel
              name: <service-name>
```
```yaml
          # Required configuration settings for the Subscriber. This is the event Sink that events are delivered to from the Channel.
          subscriber:
            ref:
              apiVersion: serving.knative.dev/v1
              kind: Service
              name: <service-name> # Configuration settings for the Subscriber. This is the event Sink that events are delivered to from the Channel.
              name: <service-name>
```

Comment (pierDipi) [reply docs/eventing/channels/subscriptions.md:98]: added also notes to signal whether a field is optional or required

**Comment by abrennan89** [comment]
@pierDipi thanks for the comments, made some updates, PTAL

**Review by pierDipi — APPROVED** [review]
/lgtm

**Pull request merged**